Fix an "invalid mutex" bug at run-time.

The mutex "&root->lock" is destroyed in the final line of sortlines().
Destroy the correct mutex (&merge_tree->lock) created by merge_tree_init()
at the start of the current block.

--- src/sort.c.orig	2012-10-23 10:14:12.000000000 -0400
+++ src/sort.c	2012-12-05 09:58:12.000000000 -0500
@@ -3918,7 +3918,7 @@
               sortlines (line, nthreads, buf.nlines, root,
                          &queue, tfp, temp_output);
               queue_destroy (&queue);
-              pthread_mutex_destroy (&root->lock);
+              pthread_mutex_destroy (&merge_tree->lock);
               merge_tree_destroy (merge_tree);
             }
           else
